### PR TITLE
Add wallet support for decimal currencies and some examples

### LINKED ISSUE #1977: Decimal support in Wallet UI

### LINKED ISSUE #1978: Denominations on the sim-chain testnet
* Keep moola+simolean for compatibility
* Bucks - pegged to USD
* agstake (anticipating integration with cosmos uagstake tokens)
* Pegasus $LINK (anticipating Peggy+Pegasus)
 
plus decimal support #1977

### PR DESCRIPTION
**Pull request by michaelfig**
Closes #1978, closes #1977

We don't actually have a path in front of the issuer, but we pretend by using `'Testnet.$USD'` and `'Testnet.$LINK'`.

The support for decimals in the wallet is genuine, though.

Screen Shot 2020-11-04 at 7 18 44 PM

### CODE REVIEW

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js [hunk 1]
```diff
@@ -158,21 +163,56 @@ export async function makeWallet({
...
+
     pursesFullState.set(
       purse,
       harden({
         ...jstate,
         purse,
         brand,
         actions: {
-          async send(receiverP, valueToSend) {
+          parseValue(str) {
```
Comment (katelynsills) [inline packages/dapp-svelte-wallet/api/src/lib-wallet.js]: I think parsing formatted values on the wallet side is probably the wrong thing. We want the decimalPlaces to only be used for display, so this parsing should be done on the dapp side. The wallet code shouldn't care about it.

**Review by katelynsills — CHANGES_REQUESTED** [review]
I think we should not be sending display-appropriate values to the wallet and parsing there. I think the UI should be responsible to both displaying and reversing the display changes such the wallet only sees nat values for fungible tokens.

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
 // @ts-check
 
+import JSON5 from 'json5';
```
Comment (FUDCo) [inline packages/dapp-svelte-wallet/api/src/lib-wallet.js]: What's up with using JSON5?  I'm already a little nervous about proliferation (where by proliferation I mean N>1) of JSON libraries.

**Review by FUDCo — APPROVED** [review]
Seems mostly OK, save my one comment about JSON5, which is less a comment about this PR and more flagging that we have a broader issue we should probably establish a policy on.

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js (hunk 2 above)
Comment (michaelfig) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js]: I'll be moving it to the frontend.  Essentially, JSON5 allows inputting JSON-like objects (the JSON5 standard) without all the nasty quoting requirements.  No observable difference in the parsed results.

Comment (erights) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js]: Discounting the quotation, is there any strong reason to introduce JSON5?

Once there's an established std, small benefits shouldn't justify leaving the std. Large benefits might and often do.

FWIW, I expect we'll be introducing a JSON library anyway, not to deviate from the JSON syntax but to do canonical serialization. So I can't claim N <= 1 purity ;)

Comment (michaelfig) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js]: > Discounting the quotation, is there any strong reason to introduce JSON5?

> FWIW, I expect we'll be introducing a JSON library anyway, not to deviate from the JSON syntax but to do canonical serialization. So I can't claim N <= 1 purity ;)

I think JSON5 is a valuable ergonomic subset between JSON and Justin.  I'd define it in terms of our PEG stack when we start to do more language stuff.  But again, this is just in the user interfaces (where we do all kinds of things that are not what we'd do in SwingSet).

Comment (michaelfig) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js]: The UI runs under SES, and JSON5 is imported after lockdown.  Only strings are passed to `JSON5.parse`.  I think this makes it a minimal exposure.

**Review by katelynsills — APPROVED** [review]
Looks good, thanks for making those changes!

**Pull request merged**